src/Symfony/Component/DependencyInjection/ConfigurationResolverTrait.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Symfony\Component\DependencyInjection;
+
+use Symfony\Component\ExpressionLanguage\Expression;
+
+trait ConfigurationResolverTrait
+{
+    protected function resolve($value)
+    {
+        if (is_string($value) && strlen($value) > 0 && $value[0] == '@') {
+            if (strlen($value) > 1 && $value[1] == '=') {
+                return new Expression(substr($value, 2));
+            } else {
```

Comment:
Actually, no. It enables you to use ```@service_name``` as a value, which will pass in the service named "service_name". I can totally see how that might be useless (especially given that if you REALLY wanted to do that, you can just use ```"@=service('service_name')"```, but this is for consistency. Because I'm using the @-syntax, I think it would be confusing if this did not work.

If we don't want this case, we should at least consider changing the syntax entirely for this particular feature.